### ISSUE TITLE
ERROR: Unknown format code 'b' for object of type 'str' on ubuntu 22.04

### ISSUE THREAD

**Issue by boukendesho** [Yubico/yubikey-manager#619]
- YubiKey Manager (ykman) version: 5.5.0
- How was it installed?: PPA
- Operating system and version: Ubuntu 22.04.4 LTS
- YubiKey model and version: YubiKey 5C NFC
- Bug description summary:

Only happen on ubuntu 22.04 and ok on ubuntu 24.04

Steps to reproduce
---

run
`
$ ykman info
`

Expected result
---

show correct info.

Actual results and logs
---

```
$ ykman info
Device type: YubiKey 5 NFC
Serial number: xxxxxxxx
Firmware version: 5.2.4
Form factor: Keychain (USB-A)
Enabled USB interfaces: OTP, FIDO, CCID
NFC transport is enabled

ERROR: Unknown format code 'b' for object of type 'str'
```

Other info
---

**Comment by pedrohdz** [comment]
I am experiencing the same after a `pip install yubikey-manager` on MacOS 14.5.

`python3.12 --version`:
```
Python 3.12.4
```

`pip freeze`:
```
cffi==1.16.0
click==8.1.7
cryptography==42.0.8
fido2==1.1.3
jaraco.classes==3.4.0
jaraco.context==5.3.0
jaraco.functools==4.0.1
keyring==25.2.1
more-itertools==10.3.0
pycparser==2.22
pyscard==2.0.10
wheel==0.43.0
yubikey-manager==5.5.0
```

Let me know if there is any other information that I can post that would be helpful.

Thanks!

**Comment by dainnilsson** [comment]
Issue confirmed. There is a compatibility with older versions of Python. A workaround is to install a newer Python version.
@pedrohdz - This should not be an issue on Python 3.12. Can you provide the output of `ykman --diagnose`?

**Comment by pedrohdz** [comment]
@dainnilsson 
Here you go:

```
ykman:            5.5.0
Python:           3.9.19 (main, Mar 20 2024, 19:50:25) 
[Clang 15.0.0 (clang-1500.1.0.2.5)]
Platform:         darwin
Arch:             arm64
System date:      2024-07-01
Running as admin: False
Detected PC/SC readers:
  Yubico YubiKey OTP+FIDO+CCID: Success

Detected YubiKeys over PC/SC:
  ScardYubiKeyDevice(pid=0407, fingerprint='Yubico YubiKey OTP+FIDO+CCID'):
    Management:  
      Raw Info: 2e0102023f0302023f020400d9188a04010105030502070602000007010f0801000d02023f0e02023e0a01000f0100
      DeviceInfo:
        config:        
          enabled_capabilities:      
            USB: OTP|U2F|FIDO2|OATH|PIV|OPENPGP: 0x23f
            NFC: U2F|FIDO2|OATH|PIV|OPENPGP: 0x23e

          auto_eject_timeout:         0
          challenge_response_timeout: 15
          device_flags:               0
          nfc_restricted:             False

        serial:         14227594
        version:        5.2.7
        form_factor:    Keychain (USB-A)
        supported_capabilities:
          USB: OTP|U2F|FIDO2|OATH|PIV|OPENPGP: 0x23f
          NFC: OTP|U2F|FIDO2|OATH|PIV|OPENPGP: 0x23f

        is_locked:      False
        is_fips:        False
        is_sky:         False
        part_number:    None
        fips_capable:   : 0x0
        fips_approved:  : 0x0
        pin_complexity: False
        reset_blocked:  : 0x0
        fps_version:    None
        stm_version:    None

      Name: YubiKey 5 NFC

    PIV:         
      PIV version:              5.2.7
      PIN tries remaining:      3
      Management key algorithm: TDES
      CHUID: No data available
      CCC:   No data available

    OATH:        
      Oath version:       5.2.7
      Password protected: True

    OpenPGP:     
      OpenPGP version:            3.4
      Application version:        5.2.7
      PIN tries remaining:        6
      Reset code tries remaining: 0
      Admin PIN tries remaining:  6
      Require PIN for signature:  Once
      KDF enabled:                False
      Touch policies:            
        Signature key:      Cached
        Encryption key:     Cached
        Authentication key: Cached
        Attestation key:    Cached

    YubiHSM Auth: YubiHSM Auth not accessible ApplicationNotAvailableError()

Detected YubiKeys over HID OTP:
  OtpYubiKeyDevice(pid=0407, fingerprint='4295003103'):
    Management:
      Raw Info: 2e0102023f0302023f020400d9188a04010105030502070602000007010f0801000d02023f0e02023e0a01000f0100
      DeviceInfo:
        config:        
          enabled_capabilities:      
            USB: OTP|U2F|FIDO2|OATH|PIV|OPENPGP: 0x23f
            NFC: U2F|FIDO2|OATH|PIV|OPENPGP: 0x23e

          auto_eject_timeout:         0
          challenge_response_timeout: 15
          device_flags:               0
          nfc_restricted:             False

        serial:         14227594
        version:        5.2.7
        form_factor:    Keychain (USB-A)
        supported_capabilities:
          USB: OTP|U2F|FIDO2|OATH|PIV|OPENPGP: 0x23f
          NFC: OTP|U2F|FIDO2|OATH|PIV|OPENPGP: 0x23f

        is_locked:      False
        is_fips:        False
        is_sky:         False
        part_number:    None
        fips_capable:   : 0x0
        fips_approved:  : 0x0
        pin_complexity: False
        reset_blocked:  : 0x0
        fps_version:    None
        stm_version:    None

      Name: YubiKey 5 NFC

    OTP:
      ConfigState(configured: (True, False), touch_triggered: (True, False), led_inverted: False)

Detected YubiKeys over HID FIDO:
  CtapYubiKeyDevice(pid=0407, fingerprint='4295003107'):
    CTAP device version:      5.2.7
    CTAPHID protocol version: 2
    Capabilities:             5
    Management:              
      Raw Info: 2e0102023f0302023f020400d9188a04010105030502070602000007010f0801000d02023f0e02023e0a01000f0100
      DeviceInfo:
        config:        
          enabled_capabilities:      
            USB: OTP|U2F|FIDO2|OATH|PIV|OPENPGP: 0x23f
            NFC: U2F|FIDO2|OATH|PIV|OPENPGP: 0x23e

          auto_eject_timeout:         0
          challenge_response_timeout: 15
          device_flags:               0
          nfc_restricted:             False

        serial:         14227594
        version:        5.2.7
        form_factor:    Keychain (USB-A)
        supported_capabilities:
          USB: OTP|U2F|FIDO2|OATH|PIV|OPENPGP: 0x23f
          NFC: OTP|U2F|FIDO2|OATH|PIV|OPENPGP: 0x23f

        is_locked:      False
        is_fips:        False
        is_sky:         False
        part_number:    None
        fips_capable:   : 0x0
        fips_approved:  : 0x0
        pin_complexity: False
        reset_blocked:  : 0x0
        fps_version:    None
        stm_version:    None

      Name: YubiKey 5 NFC

    Ctap2Info:  
      versions:                        
        U2F_V2
        FIDO_2_0
        FIDO_2_1_PRE

      extensions:                      
        credProtect
        hmac-secret

      aaguid:                           2fc0579f811347eab116bb5a8db9202a
      options:                         
        rk:                    True
        up:                    True
        plat:                  False
        clientPin:             True
        credentialMgmtPreview: True

      max_msg_size:                     1200
      pin_uv_protocols:                
        1

      max_creds_in_list:                8
      max_cred_id_length:               128
      transports:                      
        nfc
        usb

      algorithms:                      
        alg:  -7
        type: public-key
        alg:  -8
        type: public-key

      max_large_blob:                   None
      force_pin_change:                 False
      min_pin_length:                   4
      firmware_version:                 None
      max_cred_blob_length:             None
      max_rpids_for_min_pin:            0
      preferred_platform_uv_attempts:   None
      uv_modality:                      None
      certifications:                   None
      remaining_disc_creds:             None
      vendor_prototype_config_commands: None

    PIN retries: (8, None)

End of diagnostics
```

**Comment by dainnilsson** [comment]
The serial number of your YubiKey will be contained, along with the type of YubiKey and some configuration. Mainly I'm curious about the initial lines showing OS and Python version information, so it's enough to just provide that part!

**Comment by pedrohdz** [comment]
@dainnilsson 
Here you go.  Let me know if there is something missing:

```
ykman:            5.5.0
Python:           3.12.4 (main, Jun  8 2024, 04:51:36) [Clang 15.0.0 (clang-1500.1.0.2.5)]
Platform:         darwin
Arch:             arm64
System date:      2024-07-01
Running as admin: False
Detected PC/SC readers:
  Yubico YubiKey OTP+FIDO+CCID: Success
```

**Comment by dainnilsson** [comment]
@pedrohdz I'm surprized to see ykman using Python 3.12 there, as that shouldn't produce the formatting issue. Are you certain that this is the same installation of ykman that was producing the issue? Do you have other versions of Python installed?

Anyway, we will be releasing a new version shortly to deal with this.

**Comment by pedrohdz** [comment]
@dainnilsson 
I might have mixed-up versions.  It was on py3.9 at first, that gave the error for sure. I recreated the venv with py3.12, I **thought** I saw the error there. The error is gone now.  I was forced to restart my terminal and lost my scroll history.

Either way, issue resolved!

Thank you for addressing this so promptly!

**Comment by dainnilsson** [comment]
Good, glad to have confirmation that it is not an issue under Python 3.12.

So, to anyone else in the thread: We will release a fixed 5.5.1 soon, but in the meantime updating Python to 3.12 solves the issue.

**Comment by dainnilsson** [comment]
5.5.1 is now released with a fix for this, it should work on older Python versions without throwing the error.
